server/src/main/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemServicesImpl.java

```diff
@@ -1,82 +1,158 @@
+            final UUID pdlId = checkinRecord != null ? checkinRecord.getPdlId() : null;
+            validate(checkinId == null || createById == null, "Invalid agenda item %s", agendaItem);
+            validate(id == null || agendaItemRepository.findById(id).isEmpty(), "Unable to locate agenda item to update with id %s", agendaItem.getId());
+            validate(checkinRepo.findById(checkinId).isEmpty(), "CheckIn %s doesn't exist", checkinId);
+            validate(memberRepo.findById(createById).isEmpty(), "Member %s doesn't exist", createById);
+            if (!isAdmin && isCompleted) {
+                validate(!currentUser.getId().equals(pdlId) && !currentUser.getId().equals(createById), "User is unauthorized to do this operation");
             }
 
-            agendaItemRet = agendaItemRepo.update(agendaItem);
+            agendaItemRet = agendaItemRepository.update(agendaItem);
         }
         return agendaItemRet;
     }
 
-    public Set<AgendaItem> findByFields(@Nullable UUID checkinid,@Nullable UUID createdbyid) {
-        Set<AgendaItem> agendaItems = new HashSet<>(
-                agendaItemRepo.search(nullSafeUUIDToString(checkinid), nullSafeUUIDToString(createdbyid)));
 
-        return agendaItems;
+    @Override
+    public Set<AgendaItem> findByFields(UUID checkinid, UUID createbyid) {
+        String workEmail = securityService != null ? securityService.getAuthentication().get().getAttributes().get("email").toString() : null;
+        MemberProfile currentUser = workEmail != null ? currentUserServices.findOrSaveUser(null, workEmail) : null;
+        Boolean isAdmin = securityService != null ? securityService.hasRole(RoleType.Constants.ADMIN_ROLE) : false;
+
+        if (checkinid != null) {
+            CheckIn checkinRecord = checkinRepo.findById(checkinid).orElse(null);
+            final UUID pdlId = checkinRecord != null ? checkinRecord.getPdlId() : null;
+            final UUID teamMemberId = checkinRecord != null ? checkinRecord.getTeamMemberId() : null;
+            validate(!currentUser.getId().equals(pdlId) && !currentUser.getId().equals(teamMemberId) && !isAdmin, "User is unauthorized to do this operation");
+        } else if (createbyid != null) {
+            MemberProfile memberRecord = memberRepo.findById(createbyid).orElse(null);
+            validate(!currentUser.getId().equals(memberRecord.getId()) && !isAdmin, "User is unauthorized to do this operation");
+        } else {
+            validate(!isAdmin, "User is unauthorized to do this operation");
+        }
+
+        Set<AgendaItem> agendaItem = new HashSet<>(agendaItemRepository.search(nullSafeUUIDToString(checkinid), nullSafeUUIDToString(createbyid)));
+
+        return agendaItem;
     }
 
+    @Override
     public void delete(@NotNull UUID id) {
-        agendaItemRepo.deleteById(id);
-    }
-}
+        String workEmail = securityService != null ? securityService.getAuthentication().get().getAttributes().get("email").toString() : null;
```

Comment:
There are no tests exercising this method

Reply:
Added the tests for delete.